doc/website/blog/2022-04-07-how-twitter-can-implement.md

```diff
@@ -0,0 +1,113 @@
+* Schema [Visualization](https://github.com/hedwigz/entviz).
+* Monitoring w/ [sqlcommenter](https://entgo.io/blog/2021/10/19/sqlcomment-support-for-ent)
+
+### Enthistory
+Enthistory is an extension that we started developing when we wanted to add an "Activity & History" panel to one of our web services. The panel's role is to show who changed what and when (aka auditing). In [Atlas](https://atlasgo.io/), a tool for managing databases using declarative HCL files, we have an entity called "schema" which is essentially a large text blob. Any change to the schema is logged and can later be viewed in the "Activity & History" panel.
+
+![Activity and History](activity_and_history.gif)
+
+<div style={{textAlign: 'center'}}>
+  <p style={{fontSize: 12}}>The "Activity & History" screen in Atlas</p>
+</div>
+
+This feature is very common and can be found in many apps, such as Google docs, GitHub PRs, and Facebook posts, but is unfortunately missing in the very popular and beloved Twitter.
+
+Over 3 Million people voted in favor of adding the "edit button" to Twitter, so let me show you how Twitter can make their users happy without breaking a sweat!
+
+![Twitter+Edit Button](twitter_with_edit.gif)
+
+Enthistory utilizes the extensions subsystem of Ent to let the user declaratively "track" the changes of a certain field. Let's see what you'd have to do if you weren't using Enthistory: For example, consider an app similar to Twitter. It has a table called "tweets" and one of its columns is the tweet content.
+
+| id      | content | created_at | author_id |
+| ----------- | ----------- | ----------- | ----------- |
+| 1      | Hello Twitter!       | 2022-04-06T13:45:34+00:00       | 123       |
+| 2      | Hello Gophers!       | 2022-04-06T14:03:54+00:00       | 456       |
+
+Now, assume that we want to allow users to edit the content, and simultaneously display the changes in the frontend. There are several common approaches for solving this problem, each with its own pros and cons but we will dive into those in another technical post. For now, a possible solution for this is to create a table "tweets_history" which records the changes of a tweet:
+
+| id      | tweet_id | timestamp | event | content |
+| ----------- | ----------- | ----------- | ----------- | ----------- |
+| 1      | 1       | 2022-04-06T12:30:00+00:00       | CREATED       | hello world!       |
+| 2      | 2       | 2022-04-06T13:45:34+00:00       | UPDATED       | hello Twitter!       |
+
+With a table similar to the one above, we can record changes to the original tweet "1" and if requested, we can show that it was originally tweeted at 12:30:00 with the content "hello world!" and was modified at 13:45:34 to "hello Twitter!".  
+
+To implement this, we will have to change every `UPDATE` statement for "tweets" to include an `INSERT` to "tweets_history". For correctness, we will need to wrap both statements in a transaction to avoid corrupting the history.
```

Comment:
```suggestion
To implement this, we will have to change every `UPDATE` statement for "tweets" to include an `INSERT` to "tweets_history". For correctness, we will need to wrap both statements in a transaction to avoid corrupting the history in case the first statement succeeds but the subsequent one fails.
```